massa-ledger/src/ledger.rs

```diff
@@ -184,4 +195,231 @@ impl FinalLedger {
             .get(addr)
             .map_or(false, |v| v.datastore.contains_key(key))
     }
+
+    /// Get a part of the ledger
+    /// Used for bootstrap
+    /// Parameters:
+    /// * cursor: Where we stopped in the ledger
+    ///
+    /// Returns:
+    /// A subset of the ledger starting at `cursor` and of size `LEDGER_PART_SIZE_MESSAGE_BYTES` bytes.
+    pub fn get_ledger_part(
+        &self,
+        cursor: Option<LedgerCursor>,
+    ) -> Result<(Vec<u8>, LedgerCursor), ModelsError> {
+        let mut next_cursor = cursor.unwrap_or(LedgerCursor(
+            *self
+                .sorted_ledger
+                .first_key_value()
+                .ok_or_else(|| ModelsError::BufferError("Ledger empty".into()))?
+                .0,
+            LedgerCursorStep::Start,
+        ));
+        let mut data = Vec::new();
+        let amount_serializer = AmountSerializer::new();
+        for (addr, entry) in self.sorted_ledger.range(next_cursor.0..) {
+            while (data.len() as u64) < LEDGER_PART_SIZE_MESSAGE_BYTES {
```

Comment:
note that LEDGER_PART_SIZE_MESSAGE_BYTES is an estimate: it will often be overshot a bit by the last added entry

Reply:
Yes. It's will always be a bit more (see https://github.com/massalabs/massa/pull/2625#discussion_r878122100)